### ISSUE TITLE
Where does OM get the arrow direction without a magnetometer present?

### ISSUE THREAD

**Issue by emersonriver** [organicmaps/organicmaps#7971]
In a device without a built-in magnetometer/compass where does OM get the arrow direction?

This is the arrow in question:
![2](https://github.com/organicmaps/organicmaps/assets/1561878/5c3f0e08-3094-4603-b0c0-f6b49a8bef4d)

Google maps simply does not display a direction without a magnetometer, but the OM arrow is constantly switching direction. 

Possibly related to: 
[issue 6410](https://github.com/organicmaps/organicmaps/issues/6410)

**Comment by emersonriver** [comment]
Apologies for the tardy reply. [Here](https://drive.google.com/file/d/10jmSu7braSxfHawLZf0ZyQlyvksJ9Nqv/view?usp=drive_link) is the log file of an interval with seemingly random arrow directions.

**Comment by biodranik** [comment]
Logs clearly show that your device does not have any sensor:
```
04-25 17:00:58.023 (main) W/SensorHelper: SensorHelper.java:145 start(): There is no ROTATION_VECTOR sensor, requesting GEOMAGNETIC_ROTATION_VECTOR
04-25 17:00:58.023 (main) W/SensorHelper: SensorHelper.java:150 start(): There is no GEOMAGNETIC_ROTATION_VECTOR sensor, device orientation can not be calculated
```

It could be a bug that you see an arrow. A blue dot should be displayed instead. Did you ever see a blue dot instead of an arrow?

**Comment by RedAuburn** [comment]
> GoogleFusedLocationProvider location = Location[fused *  hAcc=13 et=+9m44s713ms alt=158.485 vel=0.21416129 bear=305.7459 vAcc=4 sAcc=2 bAcc=36]

looks like fusedlocationprovider is providing a bearing, maybe it's that? (particularly as `bearingAccuracy=36`)

**Comment by biodranik** [comment]
@RedAuburn you're right! If a (wrong) bearing comes from the fused provider, it is used in the core to draw an arrow.

@emersonriver Do these arrow directions make any sense to you? E.g. are they somehow valid when you rotate the device, or when you are moving in some direction?

**Comment by emersonriver** [comment]
@biodranik The arrow directions seem completely random while standing still, while moving it seems like the arrow is pointing in the direction of movement

**Comment by RedAuburn** [comment]
looks like the best fix would be to switch to the dot when using googlefusedlocation for bearing, when bearing accuracy and/or velocity is low. (and switch back to arrow when moving)

**Comment by biodranik** [comment]
@emersonriver can you please record some logs while moving (when the arrow is showing a useful direction), and send us logs using Report a Bug with a comment when exactly you saw a good arrow position?

**Comment by emersonriver** [comment]
[Here](https://drive.google.com/file/d/1MQmHcLXYCuH8oSf8OJyk3bO0Y5m5NO-P/view?usp=sharing) is a sequence of moving & arrow pointing in direction of movement.

**Comment by biodranik** [comment]
Thanks! The bearing is calculated as a difference between two GPS points, so it has better precision on the go, and doesn't work when standing still. A proper improvement should display a dot when there's no sensor and velocity is very low.

**Comment by emersonriver** [comment]
From a user perspective it might be best to remember the last moving direction for a bit longer, since the common usage is to stop and check whether you are headed the right way.

**Comment by patepelo** [comment]
1. Should we rename this Issue to make a use of it? Or move to questions as answered?

2.
>From a user perspective it might be best to remember the last moving direction for a bit longer

Now that we'll have record track, maybe we should consider using this approach #6452 like in StreetComplete. To leave a trail behind of the last 30s or so. 

IMO we should change to point with bearing as well when navigation mode is off.  See: https://github.com/orgs/organicmaps/discussions/2853#discussioncomment-7577489

Preview: 
![image](https://github.com/organicmaps/organicmaps/assets/23003127/26951c7c-092c-42b0-9a07-645cad06530e)



P.S. It seems to me that proposed ideas in Discussions aren't having as much attention as Issues.

**Comment by RageGamerBoi** [comment]
I remember this being a thing on many devices a while ago and having apps that would calculate heading based on GPS above ~10mph *even on devices with a magnetometer* because that's actually more accurate in that situation.
I believe googlefusedlocation now integrates this into Android and only falls back to also using that GPS heading below said speed if a magnetometer is not present.
I believe this is an acceptable compromise to allow for better directional accuracy in every other scenario.
